packages/cache/src/cache.ts

```diff
@@ -0,0 +1,169 @@
+        await cacheHttpClient.downloadCache(
+          cacheEntry.archiveLocation,
+          archivePath
+        )
+
+        const archiveFileSize = utils.getArchiveFileSize(archivePath)
+        core.info(
+          `Cache Size: ~${Math.round(
+            archiveFileSize / (1024 * 1024)
+          )} MB (${archiveFileSize} B)`
+        )
+
+        await extractTar(archivePath, compressionMethod)
+      } finally {
+        // Try to delete the archive to save space
+        try {
+          await utils.unlinkFile(archivePath)
+        } catch (error) {
+          core.debug(`Failed to delete archive: ${error}`)
+        }
+      }
+
+      core.info(`Cache restored from key: ${cacheEntry && cacheEntry.cacheKey}`)
+
+      return cacheEntry.cacheKey
+    } catch (error) {
+      utils.logWarning(error.message)
+      return undefined
+    }
+  } catch (error) {
+    core.setFailed(error.message)
+    return undefined
+  }
+}
+
+/**
+ * Saves a file with the specified key
+ *
+ * @param path a string representing files to be cached
+ * @param key an explicit key for restoring the cache
+ * @returns number returns cacheId if the cache was saved successfully, otherwise return -1
+ */
+export async function saveCache(path: string, key: string): Promise<number> {
+  try {
+    if (!path || path.length === 0) {
+      throw new Error('Input required and not supplied: path')
+    }
+
+    const compressionMethod = await utils.getCompressionMethod()
+
+    core.debug('Reserving Cache')
+    const cacheId = await cacheHttpClient.reserveCache(key, path, {
+      compressionMethod
+    })
+    if (cacheId === -1) {
+      core.info(
+        `Unable to reserve cache with key ${key}, another job may be creating this cache.`
+      )
+      return -1
+    }
+    core.debug(`Cache ID: ${cacheId}`)
+    const cachePaths = await utils.resolvePaths(
+      path.split('\n').filter(x => x !== '')
+    )
+
+    core.debug('Cache Paths:')
+    core.debug(`${JSON.stringify(cachePaths)}`)
+
+    const archiveFolder = await utils.createTempDirectory()
+    const archivePath = pathUtils.join(
+      archiveFolder,
+      utils.getCacheFileName(compressionMethod)
+    )
+
+    core.debug(`Archive Path: ${archivePath}`)
+
+    await createTar(archiveFolder, cachePaths, compressionMethod)
+
+    const fileSizeLimit = 5 * 1024 * 1024 * 1024 // 5GB per repo limit
+    const archiveFileSize = utils.getArchiveFileSize(archivePath)
+    core.debug(`File Size: ${archiveFileSize}`)
+    if (archiveFileSize > fileSizeLimit) {
+      utils.logWarning(
+        `Cache size of ~${Math.round(
+          archiveFileSize / (1024 * 1024)
+        )} MB (${archiveFileSize} B) is over the 5GB limit, not saving cache.`
+      )
+      return -1
+    }
+
+    core.debug(`Saving Cache (ID: ${cacheId})`)
+    await cacheHttpClient.saveCache(cacheId, archivePath)
+
+    return cacheId
+  } catch (error) {
+    utils.logWarning(error.message)
```

Comment:
We should either make this configurable or always throw an error. The package consumer can control whether they want to catch the error or allow it to fail the step.